### PR TITLE
Cb Live Response - adapt to Cb Defense

### PR DESCRIPTION
**Pull request by noaco**
As agreed - this PR will only add adaptation for the integration to use both Cb Defense and Cb Response.

Will open a different issue for the integration enhancement.

### CODE REVIEW

File: Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml [hunk 1]
```diff
@@ -13,11 +13,31 @@ configuration:
...
+  defaultvalue: ""
+  type: 8
+  required: false
+- display: API Token (Cb Response)
   name: apitoken
   defaultvalue: ""
   type: 4
-  required: true
+  required: false
+- display: Use Cb Defense
```
Comment (anara123) [inline Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml]: remove this param

File: Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml [hunk 2]
```diff
@@ -13,11 +13,31 @@ configuration:
   defaultvalue: ""
   type: 0
   required: true
-- display: API Token
+- display: Use Cb Response
```
Comment (anara123) [inline Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml]: remove this param, check in the tests that if API Token is provided then CB response, if API Key and connector provided then use CB defense

File: Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml [hunk 3]
```diff
@@ -13,11 +13,31 @@ configuration:
...
   defaultvalue: ""
   type: 4
-  required: true
+  required: false
+- display: Use Cb Defense
+  name: defense
+  defaultvalue: ""
+  type: 8
+  required: false
+- display: API Key (Cb Defense)
```
Comment (anara123) [inline Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml]: API Key (Cb Defense) => API Key (CB Defense)

File: Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml [hunk 4]
```diff
@@ -13,11 +13,31 @@ configuration:
   defaultvalue: ""
   type: 0
   required: true
-- display: API Token
+- display: Use Cb Response
+  name: response
+  defaultvalue: ""
+  type: 8
+  required: false
+- display: API Token (Cb Response)
```
Comment (anara123) [inline Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml]: API Token (Cb Response) => API Token (CB Response)

File: Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml [hunk 5]
```diff
@@ -34,11 +54,52 @@ script:
 
     /// Constants
     const SLEEP_BETWEEN_RETRIES = 5000;
+    const resource = (function() {
```
Comment (anara123) [inline Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml]: no need to wrap this function, you can just execute the code.

File: Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml [hunk 6]
```diff
@@ -34,11 +54,52 @@ script:
...
+            };
+        }
+        throw 'Please select a product';
+    })();
 
     /// Base Functions
 
     function sendRequest(path, method, requestParams, headers, ignoredStatusCodes) {
-        let baseUrl = params.serverurl + '/api';
+        let baseUrl = `${params.serverurl}/${resource.api}`;
```
Comment (anara123) [inline Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml]: I would prefer to have BASE_URL at the beginning of the code.

File: Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml [hunk 7]
```diff
@@ -78,14 +137,11 @@ script:
...
         }
-
-        //logInfo(`Result: ${result.Body}`);
-
         return result;
     }
 
     function sendFileRequest(path, fileId, requestParams, headers, ignoredStatusCodes) {
-        let baseUrl = params.serverurl + '/api';
+        let baseUrl = `${params.serverurl}/${resource.api}`;
```
Comment (anara123) [inline Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml]: Same, BASE_URL at the beginning of the code

File: Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml [hunk 8]
```diff
@@ -124,7 +180,7 @@ script:
     /// Commands
 
     function testModule() {
-        let res = sendRequest('/v1/cblr/session', 'GET');
+        let res = sendRequest(`/${resource.version}/cblr/session`, 'GET');
```
Comment (anara123) [inline Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml]: It seems that `/${resource.version}/cblr` can be part of the BASE_URL

**Comment by noaco** [comment]
@anara123 please approve [content-test-conf/pull/87](https://github.com/demisto/content-test-conf/pull/87)

**Pull request merged**